lib/resources/aws/aws_iam_policy.rb

```diff
@@ -83,7 +83,14 @@ def policy
 
   def statement_count
     return nil unless exists?
-    policy['Statement'].count
+    # Typically it is an array of statements
+    if policy['Statement'].is_a? Array
+      policy['Statement'].count    
+    else
+      # But if there is one statement, it is permissable to degenerate the array,
+      # and place the statement as a hash directly under the 'Statement' key
+      return 1
+    end
```

Comment:
I think this came up in another PR, but `Array()` is designed to return a given array, or wrap a non-array item so that you are certain to have an array. Seems like a good fit here! `Array(policy['Statement']).count`

Otherwise maybe treat the not-an-array case as a condition to guard against and sling it under the existing guard statement?
```ruby
def statement_count
  return nil unless exists?
  # AWS spec allows either a single item or an array of items
  return 1 unless policy['Statement'].respond_to?(:count)

  policy['Statement'].count
end

(The comment is just How The AWS API Always Works, right? I think this came up a few weeks ago. So I wonder if the comment is necessary/helpful.)

Reply:
I do use Array() in other places in the code, and even in this file.  It's useful.  But in this case, it does not behave as desired:
```
irb(main):001:0> h = {a: 1, b: 2}
=> {:a=>1, :b=>2}
irb(main):002:0> Array(h)
=> [[:a, 1], [:b, 2]]
```
Desired result - wrap the hash in an array: `[{:a=>1, :b=>2}]`

I do feel the comment is necessary.  The official grammar for the Policy JSON is daunting (at least to me), and without the comment, the logic would seem a bit bizarre.

Reply:
Ahhh, right, right. I didn't realize the single-item case was a hash. (See also `respond_to?(:count)`.)